agent/agent.go

```diff
@@ -606,6 +606,11 @@ func (agt *Agent) RunTask() (*apimodels.EndTaskResponse, error) {
 		grip.Debugf("stopping simple stats collector for task:", agt.GetCurrentTaskId())
 	}()
 
+	ctx, cancel := context.WithCancel(context.TODO())
+	defer cancel()
```

Comment:
We want to cancel before we finish and wait, but if the agent panics, (or we make a change that has us return early from this function) then we want to make sure that we cancel, but ideally we'll cancel on 643.